presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -948,12 +948,23 @@ std::shared_ptr<connector::ConnectorTableHandle> toConnectorTableHandle(
     const VeloxExprConverter& exprConverter,
     const TypeParser& typeParser,
     std::unordered_map<std::string, std::shared_ptr<connector::ColumnHandle>>&
-        partitionColumns) {
+        assignments) {
   if (auto hiveLayout =
           std::dynamic_pointer_cast<const protocol::HiveTableLayoutHandle>(
               tableHandle.connectorTableLayout)) {
     for (const auto& entry : hiveLayout->partitionColumns) {
-      partitionColumns.emplace(entry.name, toColumnHandle(&entry, typeParser));
+      assignments.emplace(entry.name, toColumnHandle(&entry, typeParser));
+    }
+
+    // Add synthesized columns to the TableScanNode columnHandles as well.
+    for (const auto& entry : hiveLayout->predicateColumns) {
+      if (toHiveColumnType(entry.second.columnType) ==
+          velox::connector::hive::HiveColumnHandle::ColumnType::kSynthesized) {
+        if (assignments.count(entry.first) == 0) {
+          assignments.emplace(
+              entry.first, toColumnHandle(&entry.second, typeParser));
+        }
```

Comment:
```suggestion
    for (const auto& [name, col] : hiveLayout->predicateColumns) {
      if (toHiveColumnType(col.columnType) ==
          velox::connector::hive::HiveColumnHandle::ColumnType::kSynthesized) {
        VELOX_CHECK(assignments.emplace(name, toColumnHandle(&col, typeParser)).second, "Duplicate assignment: {}", name);
```

Reply:
@Yuhta : We want to add these to the assignments list only if they are not present already. Its not quite the same code.